lib/mutant/type_registry.rb

```diff
@@ -0,0 +1,44 @@
+module Mutant
+  # Generic registry for mapping types
+  class TypeRegistry
+    include Concord.new(:contents)
+
+    # Initialize object
+    #
+    # @return [undefined]
+    def initialize
+      super({})
+    end
+
+    # Raised when the type is an invalid type
+    RegistryError = Class.new(TypeError)
```

Comment:
You could probably get by just naming this `Error`, since the name of the parent namespace already includes the term "Registry".